stem/boundary.py

```diff
@@ -0,0 +1,134 @@
+from typing import List, Dict, Any, Union, Optional
+from dataclasses import dataclass, field
+from abc import ABC, abstractmethod
+
+
+@dataclass
+class BoundaryParametersABC(ABC):
+    """
+    Abstract base class for boundary parameters
+    """
+
+    pass
+
+    @property
+    @abstractmethod
+    def is_constraint(self) -> bool:
+        """
+        Helping function to determine whether the boundary should be in the list of
+        constraints (True) or in the list of loads (False).
+        Returns:
+            bool
+        """
+        pass
+
+
+@dataclass
+class DisplacementConstraint(BoundaryParametersABC):
+    """
+    Class containing the boundary parameters for displacement constraint
+
+    Inheritance:
+        - :class:`BoundaryParametersABC`
+
+    Attributes:
+        active (List[bool]): Activate/deactivate constraint for each direction.
+        is_fixed (List[bool]): Specify if constraint is fixed for each direction.
+        value (List[float]): Displacement constraint for direction [m].
+    """
+
+    active: List[bool]
+    is_fixed: List[bool]
+    value: List[float]
+
+    @property
+    def is_constraint(self) -> bool:
+        """
+        Determines whether the boundary should be in the list of
```

Comment:
since is_constraint is a property and is always true for the displacementconstraint. Can you indicate that in the docstring. E.g.: " Property which indicates if boundary condition is a constraint. True for  DisplacementConstraint "

same for the rest of the is_constraint defs